### ISSUE TITLE
[Bug]: Game-coordinator constant disconnections.

### ISSUE THREAD

**Issue by masterofobzene** [OpenTTD/OpenTTD#10877]
### Version of OpenTTD

13.1 (and older)

### Expected result

To be able to leave the dedicated server running alone. Everyday I have to restart it or use a firewall to cut its connection to force the reconnection again since it appears to be connected but it disappears from the server list. This is a pain taking into account that the passwords for the users are not saved and restarting the server deletes the passwords (and potentially the companies as the server is configured to only leave protected companies).

### Actual result

dbg: [net] Connection with Game Coordinator lost; reconnecting...
dbg: [net] Timeout while connecting to coordinator.openttd.org:3976
dbg: [net] Connection with Game Coordinator lost; reconnecting...
dbg: [net] Timeout while connecting to coordinator.openttd.org:3976
dbg: [net] Connection with Game Coordinator lost; reconnecting...
dbg: [net] Connected to coordinator.openttd.org:3976
dbg: [net] Connected to stun.openttd.org:3975
dbg: [net] [server] Client connected from 192.168.100.3 on frame 18847617
dbg: [net] [server] Client #48 closed connection

Ports 3970-3979 are open. 
I suffer no internet cuts.

### Steps to reproduce

start a dedicated server.

**Comment by TrueBrain** [comment]
Right, you mention several things here, so let's unpack. And I will ignore the unrelated rant if you don't mind :) That is for another ticket :)

The game-coordinator's backend is fully redundant; and in no time in the last month have all instances been down. So a timeout on a connection is a bit odd. Seems to indicate your server was unable to connect to the AWS network. Bit unusual.

You mention that the server is no longer listed at some point; do I understand you right that after you forcefully disconnect the connection to the coordinator, it does recover correctly? A possible reason could be that the backend has two connections from your server: one that is timing out, and a new one. It is possible that this sometimes causes a race-condition. There is code to prevent that, but I will validate if that is sufficiently guarding this situation.

To delve in a bit further on the disconnect, we need a bit of extra information. As a general rule of thumb we always reason: if it is only one user that is experiencing network problems, it is likely to be a local issue. I know that is never nice to hear, but in general it is a good starting point. And from experience we also know it can be really hard to debug these problems; just setting expectations here. But, let's give it a go; some questions for you:

- Is this server on a wired network, or on a wireless connection?
- Could you enable timestamps in the log, and send us a new one when this happens? A log without timestamps is always a bit tricky to debug for us.
- Could you enable `-dnet=4` (or even `-dnet=6` if you dare), and whenever this happens again, send us the logs? Ideally a bit before it happens and after it happens. This gives a bit more context what the server is doing.
- Does the disconnect always happen on a set time? Or on a set interval? Any additional information here would be welcome.
- You mention you don't suffer internet cuts; how do you know? What do you use to validate that? Possibly we can extend this to probe the AWS network too.

As for the `stun` connection, it is a really short outgoing connection. It simply means your server is not reachable directly from the Internet, so it tries to NAT-punch your router; and by the looks it succeeds in that. So that is not relevant for the rest of this ticket :)

**Comment by nielsmh** [comment]
OpenTTD only receives unsolicited incoming TCP and UDP on the ports you've configured your server to run on. Those _should_ be the only ones you need to forward in your router/open in firewall.
However, the game also uses the technique called STUN to attempt to establish connections between two firewalled/NATted machines, you can read a bit more about that in [game_coordinator.md](https://github.com/OpenTTD/OpenTTD/blob/13.3/docs/game_coordinator.md). But basically, that should not affect connectivity negatively, unless you have a firewall that blocks _outgoing_ connections.

When your ISP renews your lease, does that usually result in your IP-address getting changed, or do you usually keep the same address?


On a side note, the version of OpenTTD on Steam is completely identical to any other version. It's the exact same EXE file that gets uploaded to Steam, and there's nothing coded to behave different, so the only thing you should get/not get is the automatic updates.

**Comment by 2TallTyler** [comment]
@masterofobzene Are you still experiencing this problem? Just triaging old issues that haven't been touched in a while. 🙂

**Comment by masterofobzene** [comment]
I don't host games anymore and I stopped playing this game for the time being.
